### PR TITLE
Gives minor "Life Support" to Xenomorph Nests + Xeno embryos can no longer grow in dead bodies + Other fun things

### PR DESCRIPTION
**Pull request by BurgerLUA**
## About The Pull Request

Fixes a bug where buckled people can unbuckle people from resin nests.
You can no longer escape from resin nest beds on your own.
Xeno embryos can no longer grow in dead bodies
Being buckled to Xenomorph nests now:
- Makes you immune to cold damage.
- Makes you immune to low pressure damage.
- Gives you FREE air (See: Self-respiration).
- Prevents you from entering hardcrit (Can still be killed).
- Prevents you from manually succumbing (You can still ghost or be killed).

## Why It's Good For The Game
I've talked to some people and they believe that Xenos are pretty much shafted when it comes to keeping hosts alive as the crew tend to do things like break windows in maintenance so crewmembers suffocate or die from cold damage before they burst. The funny thing is that larva actually grow in dead bodies anyways so as of now, this PR is currently useless. If this PR is merged, I will be making another PR that makes it so that xenomorph larva don't grow in dead hosts.

As for the rest of the features, they were suggested from players or developers to make the PR more balanced and fair for both sides. The funny thing is that 99% of the things in this PR aren't even my own ideas so please don't cyberbully me on reddit.

## Changelog
:cl: BurgerBB
balance: Gives minor life support to xeno beds; being buckled to a xeno bed makes you unable to succumb (but you can still ghost!)
balance: Xeno embryos can no longer grow in dead bodies, and grow about twice as fast when buckled to a resin nest. All toxin damage from embyro effects were converted to brute damage.
bugfix: Buckled mobs can no longer unbuckle things from resin nests. You can no longer escape from resin nest beds on your own.
/:cl:

Also queue the feature/fix bot throwing a fit and saying I have a score of -20 because of my feature PRs on tgmc.

### CODE REVIEW

**Comment by AlinaStarkova** [comment]
> If this PR is merged, I will be making another PR that makes it so that xenomorph larva don't grow in dead hosts.

the problem is people will still suicide to fuck the xenos even if doing so makes them DNR

**Comment by BurgerLUA** [comment]
> > If this PR is merged, I will be making another PR that makes it so that xenomorph larva don't grow in dead hosts.
> 
> the problem is people will still suicide to fuck the xenos even if doing so makes them DNR

Oh no my hubris.

**Comment by IndieanaJones** [comment]
Just make chestbursters still grow in hosts that suicide, simple.

**Comment by BurgerLUA** [comment]
> Just make chestbursters still grow in hosts that suicide, simple.

The problem with this is that there are ways to easily bypass this. Repeatedly bashing your head with a toolbox does not count as a suicide, but using the suicide verb does. There is also the problem with just asking someone else to blow your brains out for you.

The more I think about this, the more I believe that this really isn't an issue that can be solved with code. Even if you code a super intellegent AI that can detect powergaming in real-time and making it so that xenos grow or not based on if you died because of powergaming, there is nothing stopping players from just throwing themselves into the supermatter.

**Comment by Thunder12345** [comment]
Larvae already keep growing if the host dies???

**Comment by BurgerLUA** [comment]
> Larvae already keep growing if the host dies???

Yes, see the PR description.

**Comment by zxaber** [comment]
Seems fine enough to me if we're moving towards that direction. This isn't so much of a "fixing xenos being shafted", as it is "fixing people thinking xenos are getting shafted".

Ironically, in the long run this will be a nerf to xenos in cases where the infected individual escapes and happens to die elsewhere.

**Comment by jughu** [comment]
Larvas bust faster if the host is still alive
Tl;dr powercreep convience

**Comment by ExcessiveUseOfCobblestone** [comment]
This will not be merged without having both in the same PR, and even then I am not entirely sold on why thats better than the current system.

If people destroy windows and depressurize you have resin structures, they should be used here.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Above

**Comment by jughu** [comment]
Also arent xenos usually the ones breaking the windows anyways like its never the crew crew just wants to hunt xenos down meanwhile xenos break the window to escape to space for example

**Comment by Dawson1917** [comment]
>Crew tend to do things like break windows in maintenance

It's normally xenos that break windows since anybody who uses a flamethrower or starts a fire in the vicinity of an alien will leave them taking burn damage for the next >1 minute, even after they've put the fire out and moved somewhere safe, unless they stand on a space tile. With that and that they're all guaranteed to die if a mech or people in riot armor or anyone with a bomb runs into the nest, they require a quick escape route. Though the queen usually gets killed even in space because they're way slower than any person or mech. 

>If this PR is merged, I will be making another PR that makes it so that xenomorph larva don't grow in dead hosts.

I think a lot needs to be addressed before that change can result in anything but making xenos even weaker.
Xenomorphs really need to get their ability to drag people through vents returned, so they can try to stealth instead of every abduction requiring you to drag a body through the hallways directly to your nest.

**Comment by BurgerLUA** [comment]
> This will not be merged without having both in the same PR, and even then I am not entirely sold on why thats better than the current system.
> 
> If people destroy windows and depressurize you have resin structures, they should be used here.

Do you have any ideas on how to PR is more mergeable? Other than what you requested, of course, because I'm implementing that now.

**Comment by BurgerLUA** [comment]
I was told I should at least try to be convincing, so here is my argument.

Currently, I believe, that the idea that a larva can grow inside a dead patient is really against good gameplay design. The reason why this was added in the first place is because people would powergame and immediately kill themselves or suicide just to fuck over xenos, even if it makes them DNR. The thing is that even with this feature currently implemented, a powergamer could find plenty of ways to deny a xeno burst without using medical help. They could:

- Throw themselves into the supermatter.
- Throw themselves into the singularity.
- Disembowel themselves, removing all organs (including the egg) and still keeping them "revivable".
- Space themselves (Unreliable as you could drift back to the station, but you're still made revivable).
- Use the suicide verb with an oxygen tank as it would gib you.
- Get their DNA instability to absurd levels and hope they get dusted/gibbed.
- Chef.
- Flee to lavaland and throw themselves in lava.
- Obtain the Internal Martydom mutation and die.
- Activate a shield wall while in the middle of it.
- Use the suicide verb with an RCD.
- Use the suicide verb with a skub.
- Use the suicide verb with an explosive spear.
- Use the suicide verb with a monkey cube.
- Click on an Arnold Pizza with a Pineapple Pizza.
- Call in a supply pod that can gib you (Cloak delivery does this, for example).
- Use the suicide verb with watermelon seeds.
- Increase your blood level to insane levels.
- Bombs.
- Become fat and consume minttoxin.
- Consume at least 50u of monkey powder and some water.
- Use the suicide verb with an anomaly core.
- Inject plasma into an oil slime core.
- Ashwalker sacrifice.
- Overuse the cursed slot machine.
- Shuttlegib.
- Be unworthy when you use the suicide verb with the janitor cart.
- Get drilled while dead with a ripley.

To put it short, there are plenty of ways to kill yourself and the xeno baby inside you, even with this present. What currently exists is trying to restricting something that could not be restricted in the first place and thus affects the balance of xenos in a way that doesn't really make any sense. This PR changes that and balances to levels that make a little more sense.

**Comment by jughu** [comment]
Just pr it on skyrats

**Comment by jughu** [comment]
https://github.com/Skyrat-SS13/Skyrat-tg/pull/7949

Your playing on a RP server nobody will litteraly do any of these things you just mentioned hell we don't even do it here

**Comment by BurgerLUA** [comment]
> [Skyrat-SS13/Skyrat-tg#7949](https://github.com/Skyrat-SS13/Skyrat-tg/pull/7949)
> 
> Your playing on a RP server nobody will litteraly do any of these things you just mentioned hell we don't even do it here

exactly which is why I'm trying to make it here

**Comment by jughu** [comment]
We don't have those issues you mentioned
Beside before you go more into this change you know if people are nested to eachother they can just unbuckle eachother right?

**Comment by BurgerLUA** [comment]
> We don't have those issues you mentioned
> Beside before you go more into this change you know if people are nested to eachother they can just unbuckle eachother right?

They should not be able to, and if so, that's a bug. Nested people are given the TRAIT_HANDS_BLOCKED trait which should prevent this.

**Comment by jughu** [comment]
This is intended functionality auctally
The trait hands blocked purpose only exists so you cant grab a lasergun and shoot em

**Comment by BurgerLUA** [comment]
> This will not be merged without having both in the same PR, and even then I am not entirely sold on why thats better than the current system.
> 
> If people destroy windows and depressurize you have resin structures, they should be used here.

Yes hello it's ready for review. Lots of other things were added on top after discussions on the /tg/ discord.

**Comment by AlinaStarkova** [comment]
> 
> 
> I was told I should at least try to be convincing, so here is my argument.
> 
> Currently, I believe, that the idea that a larva can grow inside a dead patient is really against good gameplay design. The reason why this was added in the first place is because people would powergame and immediately kill themselves or suicide just to fuck over xenos, even if it makes them DNR. The thing is that even with this feature currently implemented, a powergamer could find plenty of ways to deny a xeno burst without using medical help. They could:
> 
>     * Throw themselves into the supermatter.
> 
>     * Throw themselves into the singularity.
> 
>     * Disembowel themselves, removing all organs (including the egg) and still keeping them "revivable".
> 
>     * Space themselves (Unreliable as you could drift back to the station, but you're still made revivable).
> 
>     * Use the suicide verb with an oxygen tank as it would gib you.
> 
>     * Get their DNA instability to absurd levels and hope they get dusted/gibbed.
> 
>     * Chef.
> 
>     * Flee to lavaland and throw themselves in lava.
> 
>     * Obtain the Internal Martydom mutation and die.
> 
>     * Activate a shield wall while in the middle of it.
> 
>     * Use the suicide verb with an RCD.
> 
>     * Use the suicide verb with a skub.
> 
>     * Use the suicide verb with an explosive spear.
> 
>     * Use the suicide verb with a monkey cube.
> 
>     * Click on an Arnold Pizza with a Pineapple Pizza.
> 
>     * Call in a supply pod that can gib you (Cloak delivery does this, for example).
> 
>     * Use the suicide verb with watermelon seeds.
> 
>     * Increase your blood level to insane levels.
> 
>     * Bombs.
> 
>     * Become fat and consume minttoxin.
> 
>     * Consume at least 50u of monkey powder and some water.
> 
>     * Use the suicide verb with an anomaly core.
> 
>     * Inject plasma into an oil slime core.
> 
>     * Ashwalker sacrifice.
> 
>     * Overuse the cursed slot machine.
> 
>     * Shuttlegib.
> 
>     * Be unworthy when you use the suicide verb with the janitor cart.
> 
>     * Get drilled while dead with a ripley.
> 
> 
> To put it short, there are plenty of ways to kill yourself and the xeno baby inside you, even with this present. What currently exists is trying to restricting something that could not be restricted in the first place and thus affects the balance of xenos in a way that doesn't really make any sense. This PR changes that and balances to levels that make a little more sense.

except none of those ways are doable if your trapped in a xeno nest, the issue is people can suicide while in a xeno nest (theres a fancy helpfull verb in your OOC tab called "ghost" that lets you instant suicide no matter what)

**Comment by optimumtact** [comment]
> [Skyrat-SS13/Skyrat-tg#7949](https://github.com/Skyrat-SS13/Skyrat-tg/pull/7949)
> 
> Your playing on a RP server nobody will litteraly do any of these things you just mentioned hell we don't even do it here

This is not an argument you droolie, if you're going to comment on a PR here, make it about this server, so help me god I will turn you into a pretzel

**Comment by ExcessiveUseOfCobblestone** [comment]
> https://github.com/tgstation/tgstation/pull/61227#issuecomment-912940876

I mean sure they can kill themselves plenty ways now but many of those are extremely situational and the clock is constantly ticking down (and more importantly you literally perma removal urself and your ticket to get back in as antag). Whats more reliable is to simply just get a toolbox and a bedsheet and ghetto surgery it out so you can still play the game while also being a person who can help fend off the threat.

Compare that to the design you would like to head to which is "simply dying means that you cannot grow xenomorphs", now literally anywhere regardless of any tools at any point can someone deny xenos unless theyre restrained, which already prevents people from doing ANY of the ones you listed now. If the design problem is malicious players, then this PR or even the followup will not help remedy that issue.

I think its a necessary balance measurement to enable people who die either by themselves or through a friend killing them, while still punishing xenos who kill people to make them more submissive (increased growth timer). The more realistic scenario than btfoing yourself instead of just going to get surgery (tbqh I think most people would prefer to be a xeno anyways so they have a higher chance of getting the role) would be that this PR makes it easier to keep people alive, thereby making it easier for xenos to curb stomp without regard to approach (you can literally make space nests if this was merged).

> https://github.com/tgstation/tgstation/pull/61227#issuecomment-912997950

Catatonic people count as alive vessels versus suicide which count as dead ones for the purpose of xeno growth unless im mistaken.

_____

Not sold on the direction, so ill give response time for both you and any maintainer who is more interested than my disinterest in the PR before closing.

**Comment by ExcessiveUseOfCobblestone** [comment]
> > This will not be merged without having both in the same PR, and even then I am not entirely sold on why thats better than the current system.
> > If people destroy windows and depressurize you have resin structures, they should be used here.
> 
> Yes hello it's ready for review. Lots of other things were added on top after discussions on the /tg/ discord.

ping me on discord (I recently joined skyrat if ur adverse to associating with LRP TG) @ cobby#0002 instead, my github notifications drown in code owner pings
